### ISSUE TITLE
`common` does not work with `flag_alias`

### ISSUE THREAD

**Issue by luispadron** [bazelbuild/bazel#20081]
### Description of the bug:

Using the new `common` in a .bazelrc with `flag_alias` returns the following error:

```sh

ERROR: /Users/<user>/Development/cash-ios/.bazelrc: "common --flag_alias=build_config=//:build_config" disallowed. --flag_alias only supports the "build" command.
```

### Which category does this issue belong to?

CLI, Configurability

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

- Create a custom `config_setting`
- Create a .bazelrc
- Attempt to add a flag alias with the `common` command

```bzl
common --flag_alias=release_variant=//:release_variant
```

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

release 6.3.2

### Any other information, logs, or outputs that you want to share?

Not entirely sure if this is expected but it feels like its not given how `common` works for other flags.

**Comment by katre** [comment]
Which commands besides `build` (and its logical descendents, like `test` and `cquery`) do you think need support for `--flag_alias`?

**Comment by luispadron** [comment]
> Which commands besides `build` (and its logical descendents, like `test` and `cquery`) do you think need support for `--flag_alias`?

For our use case, none and I'm not sure where it might be useful in other commands. I was just basing this off of how `common` seems to be working for us with other flags. We've been able to replace usage of `build` in bazelrc with `common` except for this case.

**Comment by katre** [comment]
The code that checks this is [in `BlazeOptionHandler`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/runtime/BlazeOptionHandler.java;drc=8833fd75d54c1763c221f1ed9d2e3f24ccfcc3d8;l=616), and the tests are in [`starlark_configurations_test.sh`](https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/integration/starlark_configurations_test.sh;drc=4dc0d119a6f4901aa8353e75941c19205542743e;l=643).

The test makes it look like this is deliberate, and @gregestren added the check that it isn't supported for `test`, so I'm going to assign to him for further triage.

**Comment by katre** [comment]
I'm also re-classifying this as a Feature Request instead of a Bug, since the behavior seems to be intentional.

**Comment by luispadron** [comment]
Thats fair, I was considering this a bug based on this: https://github.com/bazelbuild/bazel/issues/3054

**Comment by gregestren** [comment]
https://github.com/bazelbuild/bazel/commit/329b22a980ad7a0835c53938afe06a111f40ac96 is the original change.

The intention was to not allow flag alias definitions for subsets of `build` (like `test`). Motivated specifically by CI systems that start with a `$ bazel canonicalize-flags` to vet flags. `canonicalize-flags` itself is a subset of `build` and any setting has to be reflected there 

`common` goes the other way up the inheritance tree and sounds safe to me.

**Comment by gregestren** [comment]
This is basically a lowest common ancestor problem.

We can only apply flag alias definitions on nodes in `canonicalize-flags`'s ancestor path.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.0.0 RC5](https://releases.bazel.build/7.0.0/rc5/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
